### ISSUE TITLE
Add project to workspace

### ISSUE THREAD

**Issue by SheepDomination** [Kruemelkatze/vscode-dashboard#39]
When switching between projects from dashboard, is it not possible to keep the project open within explorer, rather then switching or atleast give an option ?

**Comment by Kruemelkatze** [comment]
You mean the same behavior as by using  "File >> Add Folder to Workspace"?

**Comment by SheepDomination** [comment]
Yes.

**Comment by Kruemelkatze** [comment]
This is actually on my todo list. Maybe for v2.1

**Comment by SheepDomination** [comment]
Once this is added, then if the workspace is listed in explorer; all dashboard has to do is switch between the workspace.  Then when switching between the workspace, dashboard could prompt the user to keep all the workspaces open in explorer or close some.

**Comment by Kruemelkatze** [comment]
As I said a couple of times, the dashboard will not provide any additional workspace or folder management outside of what vscode provides out of the box. Due to both technical reasons (the API is quite limited for these functions) and the intended scope for the extension.

So in this context, there will be a "Add project to workspace" option, which is just a short-cut to "File >> Add folder to workspace".

**Comment by SheepDomination** [comment]
The API doesn't allow you to add the dashboard workspace automatically when opened as so it's visible in explorer ?

**Comment by Kruemelkatze** [comment]
> The API doesn't allow you to add the dashboard workspace automatically when opened as so it's visible in explorer ?

I don't know what you mean by "dashboard workspace".

**Comment by SheepDomination** [comment]
> > The API doesn't allow you to add the dashboard workspace automatically when opened as so it's visible in explorer ?
> 
> I don't know what you mean by "dashboard workspace".

The workspaces that live within dashboard.  Each project is a workspace.

**Comment by Kruemelkatze** [comment]
I'm still not quite sure. But I built an "Add project to workspace" function, that will be shipped with 2.1.0 in the next few days.

The option can be used by right-clicking on a project and will open the project additionally in the current vscode instance. So if you had Project A open and add Project B to the workspace, you will have A and B in the explorer.

You can play around with the preview version if you want to. It may still be buggy though.

[vscode-dashboard-2.1.0-preview.vsix.zip](https://github.com/Kruemelkatze/vscode-dashboard/files/4494832/vscode-dashboard-2.1.0-preview.vsix.zip)

**Comment by SheepDomination** [comment]
This is excellent, except when appending projects to the workspace the combined workspaces have a *untitled* main workspace name; how about prompting the user for a name then if the user chooses, this new combined workspaces can be saved as a project to the dashboard.

Also when appending projects to the workspace only the first project name is added to the workspace the rest are added as a CSS rule; but I know that this is still in progress.

**Comment by Kruemelkatze** [comment]
> This is excellent, except when appending projects to the workspace the combined workspaces have a _untitled_ main workspace name; how about prompting the user for a name then if the user chooses, this new combined workspaces can be saved as a project to the dashboard.

This can already be achieved by using the native VSCode function "File >> Save Workspace as...". You have to add the project manually afterward, as there's no reasonable way for the extension to hook into after saving the workspace file. And I will not include any custom functionality just for this use-case.

> Also when appending projects to the workspace only the first project name is added to the workspace the rest are added as a CSS rule; but I know that this is still in progress.

I neither can reproduce the problem nor do I understand what CSS has to do with the workspace. 

When I had a folder "Thesis" open and add "vscode-dashboard" to the workspace (via the new function), I get this:

![image](https://user-images.githubusercontent.com/5564731/79700817-dfd2de80-8298-11ea-9f14-1f728bb764f7.png)

The workspace is untitled as described above, yes, but every folder has its name. Even if I add a workspace project (.code-workspace as path) to the current workspace, every opened folder is named.

**Comment by SheepDomination** [comment]
Each project name is not added to the workspace as the project name.
![cssprojectname](https://user-images.githubusercontent.com/6827459/79703094-8b6b3700-8277-11ea-9417-16b4c57ae17d.JPG)

**Comment by Kruemelkatze** [comment]
I highly suspect this is your customization code that you put into the project's name, as ```<div style='white';>``` is not valid HTML.  But that's not the problem. Your customizing "to have it exactly as you want" is the problem.

Seriously. I assumed you to be a developer. And as such, I would expect you to have at least some tendency to solve your problems on your own. All of these customizing and devtools-do-not-open problems could have been resolved by 5min of googling. Github issues aren't your private forum for getting help on customizing/misusing tools (by looking at your tickets in other repos).

I'm happy that you use my extension and I will gladly take feedback or suggestions! But please, spare me such discussions.

**Comment by SheepDomination** [comment]
What, I'm only telling you what I showed in the screen shot isn't showing the project name ?

**Comment by Kruemelkatze** [comment]
> Regardless that I customized, there is probably a solution around this.

Probably. But I'm not the one responsible for finding it.

**Comment by Kruemelkatze** [comment]
The latest v2.1.0 release now includes the discussed "Add project to workspace" option. Thanks for suggesting.

**Comment by SheepDomination** [comment]
The JSON validates, but rather then the Project Name, I continue to get the inline CSS style ?

**Comment by Kruemelkatze** [comment]
The JSON Validation per se has nothing to do with the HTML. Anyway: When testing this behavior, I also noticed a small race condition when editing the projects manually and fixed this with a new 2.1.1 version. Additionally, I included a simple html-tag-stripping before adding the project to the workspace.

So instead of displaying
```html
<div class="white"><svg ....></svg> TEST </div>
```

it now only displays ```TEST``` in the workspace.  Note that this only works on valid HTML and maybe not for all cases. 

![image](https://user-images.githubusercontent.com/5564731/80593747-04aa1d00-8a22-11ea-8583-b7ad7f5afe4d.png)
